android/src/com/frostwire/android/gui/fragments/BrowsePeerFragment.java

```diff
@@ -481,13 +508,22 @@ protected boolean onItemLongClicked(View v) {
     private void onToolbarMenuSelectModeCheckboxClick() {
         selectAllModeOn = !selectAllModeOn;
         enableSelectAllMode(selectAllModeOn, selectAllModeOn);
+        if (selectAllModeOn) {
+            selectionModeCallback.onItemChecked(getActivity(), adapter.getCheckedCount());
+        }
     }
 
     private void enableSelectAllMode(boolean selectAll, boolean autoCheckAll) {
         selectAllModeOn = selectAll;
         selectAllCheckboxContainer.setVisibility(selectAllModeOn && adapter.getCount() > 0 ? View.VISIBLE : View.GONE);
         adapter.setCheckboxesVisibility(selectAllModeOn);
         selectAllCheckbox.setChecked(autoCheckAll);
+
+        if (selectAllModeOn) {
+            ((MainActivity) getActivity()).startSupportActionMode(selectionModeCallback);
```

Comment:
I imagine the cast to `MainActivity` is not necessary, but it's not important now